tls/s2n_signature_algorithms.c

```diff
@@ -154,19 +154,6 @@ int s2n_choose_sig_scheme_from_peer_preference_list(struct s2n_connection *conn,
         GUARD(s2n_choose_default_sig_scheme(conn, &chosen_scheme));
     }
 
-    /* In TLS 1.3, SigScheme also defines the ECDSA curve to use (instead of reusing whatever ECDHE Key Exchange curve
-     * was negotiated). In TLS 1.2 and before, chosen_scheme.signature_curve must *always* be NULL, if it's not, it's
-     * a bug in s2n's preference list. */
-    S2N_ERROR_IF(conn->actual_protocol_version <= S2N_TLS12 && chosen_scheme.signature_curve != NULL, S2N_ERR_INVALID_SIGNATURE_SCHEME);
-
-    /* If TLS 1.3 is negotiated, then every ECDSA SigScheme must also define an ECDSA Curve *except* ECDSA_SHA1, which
-     * uses the same curve negotiated in the ECDHE SupportedGroups Extension. */
-    S2N_ERROR_IF(conn->actual_protocol_version == S2N_TLS13
-            && chosen_scheme.sig_alg == S2N_SIGNATURE_ECDSA
-            && chosen_scheme.hash_alg != S2N_HASH_SHA1
-            && chosen_scheme.signature_curve == NULL,
-            S2N_ERR_ECDSA_UNSUPPORTED_CURVE);
-
```

Comment:
I remove these checks because they don't cover all possible invalid signature scheme selections, and we really need to make the checks BEFORE we choose a signature scheme, not after.

Reply:
we can use `S2N_IMPLIES` now, thanks to @danielsn 

should these be moved to `s2n_init` so that we ensure that they are setup correctly?

i wouldnt block this PR on it though.

Reply:
I've got some suggestions that are better than the "check in a unit test" here: https://github.com/awslabs/s2n/issues/1606 If we're going to check in s2n_init, I'd rather we actually just set the maximum programmatically in s2n_init.